### ISSUE TITLE
[Bug] pca.GetAccountsAsync() decides to include another account

### ISSUE THREAD

**Issue by RhomGit** [AzureAD/microsoft-authentication-library-for-dotnet#2699]
**Which version of MSAL.NET are you using?**
Microsoft.Identity.Client v4.31.0

**Platform**
.NET 4.6.1 WPF Desktop

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Silent 
 
**Is this a new or existing app?**
b. The app is in production, I haven't upgraded MSAL, but started seeing this issue.

**Repro**
https://github.com/RhomGit/MSALDesktopRepro
`var accounts = await pca.GetAccountsAsync();`

**Expected behavior**
Removing all accounts and/or deleting msalcache.bin3 file would remove all accounts.
Subsequent logins **only** would be remembered.

**Actual behavior**
After an unspecified period of time, maybe an hour, GetAccountsAsync() return multiple accounts for some reason, despite the second account never being used by the app.

**Additional context / logs / screenshots**
The account that is being added isn't my Windows user, nor is it any of my logged in Office365 accounts/Edge browser/Visual Studio accounts. The only app logged in with this account is Teams? The timing seems to coincide with the very recent Windows 10 update.

This occurs in our production app and also occurs in the repro app linked above.
I'd say maximum 7 days since I noticed it, probably less and I auth via MSAL every day.

**Comment by RhomGit** [comment]
cc: @bgavrilMS Sorry for messing up: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2683

**Comment by bgavrilMS** [comment]
@RhomGit - could you email me your app coordinates (client_id, tenant, b2c hostname etc)? bogavril    at microsoft com
And also could you clarify what accounts are used to login? Are these local accounts or Facebook / Google / other accounts?

**Comment by RhomGit** [comment]
Done. Azure B2C for anyone following this.

**Comment by bgavrilMS** [comment]
Thanks, I got the email. I tried the first steps.

The strange things is that the response does not contain an **access token**. It only contains an id_token. 

![image](https://user-images.githubusercontent.com/12273384/121891392-4889df80-cd13-11eb-9b13-859c49191449.png)

This allows your app to log in the user, but should not be used to access protected APIs. What are these scopes that you are asking permissions for? Are these a protected API of your own? Or are you just trying to log in the user?

`https://mytenant.onmicrosoft.com/read`
`https://mytenant.onmicrosoft.com/write`



Aside: MSAL wasn't really setup for this scenario, and when you AcquireTokenSilent MSAL will always perform the `refresh_token` grant. MSAL always expects an AT to be present in the response. 

Aside2: can't see 2 accounts though in the cache, I'll wait for some time.

CC @jennyf19 @jmprieur  in case you've seen this setup before
